### PR TITLE
Initial tinkerbell provider skeleton

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*

*Description of changes:*
Most of the tinkerbell provider functions need implementation, I have added TODOs for most of them

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	values := map[string]interface{}{
+		"clusterName":                  clusterSpec.ObjectMeta.Name,
+		"controlPlaneEndpointIp":       clusterSpec.Spec.ControlPlaneConfiguration.Endpoint.Host,
+		"controlPlaneReplicas":         clusterSpec.Spec.ControlPlaneConfiguration.Count,
+		"controlPlaneSshAuthorizedKey": controlPlaneMachineSpec.Users[0].SshAuthorizedKeys,
+		"controlPlaneSshUsername":      controlPlaneMachineSpec.Users[0].Name,
+		"eksaSystemNamespace":          constants.EksaSystemNamespace,
+		"format":                       format,
+		"kubernetesVersion":            bundle.KubeDistro.Kubernetes.Tag,
+		"kubeVipImage":                 bundle.VSphere.KubeVip.VersionedImage(),
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/tinkerbell.go:294]:
```suggestion
		"kubeVipImage":                 "ghcr.io/kube-vip/kube-vip:latest",
```

File: pkg/providers/tinkerbell/config/template-md.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  template:
+    spec:
+      joinConfiguration:
+        nodeRegistration:
+          kubeletExtraArgs:
+            provider-id: PROVIDER_ID
+      users:
+      - name: {{.workerSshUsername}}
+        sshAuthorizedKeys:
+        - '{{.vsphereWorkerSshAuthorizedKey}}'
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/config/template-md.yaml:58]:
```suggestion
        - '{{.workerSshAuthorizedKey}}'
```

File: pkg/constants/constants.go [hunk 3]
```diff
@@ -22,7 +22,8 @@ const (
 	EtcdAdmBootstrapProviderName            = "bootstrap-etcdadm-bootstrap"
 	EtcdadmControllerProviderName           = "bootstrap-etcdadm-controller"
 
-	VSphereProviderName = "vsphere"
-	DockerProviderName  = "docker"
-	AWSProviderName     = "aws"
+	VSphereProviderName   = "vsphere"
+	DockerProviderName    = "docker"
+	AWSProviderName       = "aws"
+	TinkerellProviderName = "tinkerbell"
```
Comment (abhinavmpandey08) [inline pkg/constants/constants.go:28]:
```suggestion
	TinkerbellProviderName = "tinkerbell"
```

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+	return []bootstrapper.BootstrapClusterOption{bootstrapper.WithEnv(env)}, nil
+}
+
+func (p *tinkerbellProvider) BootstrapSetup(ctx context.Context, clusterConfig *v1alpha1.Cluster, cluster *types.Cluster) error {
+	// TODO: figure out if we need something else here
+	return nil
+}
+
+func (p *tinkerbellProvider) Name() string {
+	return constants.TinkerellProviderName
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/tinkerbell.go:78]:
```suggestion
	return constants.TinkerbellProviderName
```

**Review by pokearu — APPROVED** [review]
/lgtm

**Pull request merged**